scripts/colab_helper.py

```diff
@@ -0,0 +1,142 @@
+# If running on Google Colab, install Pyomo and Ipopt via IDAES
+if "google.colab" in sys.modules:
+    !wget "https://raw.githubusercontent.com/idaes-pse/main/scripts/colab_helper.py"
+    import colab_helper
+    colab_helper.install_idaes()
+    colab_helper.install_ipopt()
+```
+
+For testing purposes, you may need to use:
+"https://raw.githubusercontent.com/adowling2/idaes-pse/colab-install-script/scripts/colab_helper.py"
+
+"""
+
+import shutil
+import sys
+import os.path
+import os
+import re
+
+import subprocess
+
+
+def _check_available(executable_name):
+    """Utility to check in an executable is available"""
+    return shutil.which(executable_name) or os.path.isfile(executable_name)
+
+
+def package_available(package_name):
+    """Utility to check if a package/executable is available
+
+    This supports customization, e.g., glpk, for special package names
+    """
+
+    if package_name == "glpk":
+        return _check_available("gpsol")
+    else:
+        return _check_available(package_name)
+
+
+def on_colab():
+    """Utility returns True if executed on Colab, False otherwise"""
+    return "google.colab" in sys.modules
+
+
+def install_idaes():
+    """Installs latest version of IDAES-PSE via pip"""
+
+    try:
+        import idaes
+
+        print("idaes was found! No need to install.")
+    except ImportError:
+        print("Installing idaes via pip...")
+        subprocess.run(
+            [sys.executable, "-m", "pip", "install", "-q", "idaes_pse"], check=True
+        )
+        print("idaes was successfully installed")
+        v = subprocess.run(
+            ["idaes", "--version"], check=True, capture_output=True, text=True
+        )
+        print(v.stdout)
+        print(v.stderr)
+
+
+def install_ipopt(try_conda_as_backup=False):
+    """Install Ipopt and possibly other solvers.
+
+    If running on Colab, this will install Ipopt, k_aug, and other COIN-OR
+    solvers via idaes get-extensions.
+
+    As an optional backup, this will install Ipopt via conda
+    """
+
+    # Check if Ipopt (solver) is available. If not, install it.
+    if not package_available("ipopt"):
+        print("Running idaes get-extensions to install Ipopt, k_aug, and more...")
+        subprocess.run(["idaes", "get-extensions"], check=True)
+        _update_path()
+        print("Checking solver versions:")
+        _print_solver_versions()
+
+    # Check again if Ipopt is available. If not, try conda
+    if try_conda_as_backup and not package_available("ipopt"):
+        print("Installing Ipopt via conda...")
+        subprocess.run(
+            [sys.executable, "-m", "conda", "install", "-c", "conda-forge", "ipopt"],
+            check=True,
+        )
+        print("Checking ipopt version:")
+        _print_single_solver_version("ipopt")
+
+
+def _update_path():
+    """Add idaes executables to PATH"""
+    if not re.search(re.escape(":/root/.idaes/bin/"), os.environ["PATH"]):
+        os.environ["PATH"] += ":/root/.idaes/bin/"
```

Comment:
This is a minor nit, but we probably want to put `idaes/bin` first and not last in the path (the idea being that we would be overriding anything that was previously installed into the system)

Reply:
I fixed this and tested it works on Colab.